### ISSUE TITLE
VCS Always trigger runs

### ISSUE THREAD

**Issue by Spritekin** [leg100/otf#582]
When using a VCS repo, I want my workspace to trigger runs only if a file in a folder of my repository changes (i.e. file trigger).

In TFC the Workspace creation data structure attribute file-triggers-enabled has a default value of "true" and reads: 
```
Whether to filter runs based on the changed files in a VCS push. If enabled, it uses either trigger-prefixes in conjunction with working_directory or trigger-patterns to describe the set of changed files that will start a run. If disabled, any push will trigger a run.
```
https://developer.hashicorp.com/terraform/cloud-docs/api-docs/workspaces#create-a-workspace

However in OTF it seems like it always behaves as the value is "False". Indeed it is false even when I force a value of True.

My data structure is:
```
    payload = {
      "data": {
        "attributes": {
          "name": workspace_name,
          "auto-apply": True,
          "resource-count": 0,
          "terraform-version": terraform_version,
          "working-directory": working_directory,
          "file-triggers-enabled": True,                            <<<<<<<< HERE
          "trigger-prefixes": [f"/{working_directory}"],    <<<<<<<< AND HERE
          "vcs-repo": {
            "identifier": repo_name,
            "oauth-token-id": vcs_repo_oauth_token,
            "branch": branch
          }
        },
        "type": "workspaces"
      }
    }
```

However when the workspace is created, the trigger still reads:

<img width="889" alt="Screenshot 2023-09-01 at 2 07 03 am" src="https://github.com/leg100/otf/assets/13467740/5a745b76-410a-46dd-94f0-f9dec6225f79">

In other words, the value as if the file triggers attribute was set to False.

**Comment by Spritekin** [comment]
Ah, reading the code... trigger-prefixes is not used in OTF. Also there is this "always-trigger" attribute which is not part of TFC. Anyway I adjusted my payload to:
```
   payload = {
      "data": {
        "attributes": {
          "name": workspace_name,
          "auto-apply": True,
          "resource-count": 0,
          "terraform-version": terraform_version,
          "working-directory": working_directory,
          "file-triggers-enabled": True,
          "always-trigger": False,
          "trigger-patterns": [f"/{working_directory}/*.tf"],
          "vcs-repo": {
            "identifier": repo_name,
            "oauth-token-id": vcs_repo_oauth_token,
            "branch": branch
          }
        },
        "type": "workspaces"
      }
    }
 
```

And now it worked:
<img width="897" alt="Screenshot 2023-09-01 at 2 50 57 pm" src="https://github.com/leg100/otf/assets/13467740/a4acc205-8047-4c57-98ac-1132cca5a769">

**Comment by Spritekin** [comment]
Working but not the same as in TFC.

**Comment by leg100** [comment]
Yes this does deviate from TFC. I didn't see the point in implementing trigger prefixes because trigger patterns are more powerful and permit setting prefix as well.

But yes this is not clear to anyone using the API. It would be good to document the API and to note where there are discrepancies. This is no small task because the API is vast and there are a fair few discrepancies.
